packages/dashboard/src/components/pageStructure/index.js

```diff
@@ -164,6 +194,28 @@ export function LeftRail() {
     trackClick(evt, path.trackingEvent);
   }, []);
 
+  // See how many templates are new based on the current date
+  useEffect(() => {
+    let mounted = true;
+    getTemplateCreationDates().then((templateEntries) => {
+      if (mounted) {
+        const currentDate = toDate(new Date(), getOptions());
+        setNumNewTemplates(
+          templateEntries
+            .map(([, creationDate]) => toDate(creationDate, getOptions()))
+            .map((creationDate) =>
+              differenceInSeconds(currentDate, creationDate)
+            )
+            .filter((deltaSeconds) => deltaSeconds < SIXTY_DAYS_IN_SECONDS)
+            .length
+        );
+      }
+    });
+    return () => {
+      mounted = false;
+    };
+  }, []);
```

Comment:
`getTemplateCreationDates()` is not really useful beyond this use case of getting this count.

Since you've already added `@web-stories-wp/date` as a dependency anyway, it might make more sense to move more of the logic from here to the templates package, turning it into a more general `getRecentTemplateNames()` function or something. Maybe something like this (untested);

```js
async function getRecentTemplateNames(days) {
  const currentDate = toDate(new Date(), getOptions());

  const templatesWithDates = await Promise.all(
    TEMPLATE_NAMES.map((slug) =>
      import(
        /* webpackChunkName: "chunk-web-stories-template-[index]-creationDate" */ `./raw/${slug}/creationDate`
      ).then((data) => ({ slug, date: toDate(data.default, getOptions()) }))
    )
  );

  return templatesWithDates
    .filter(({ date }) => differenceInDays(currentDate, date) <= days)
    .map(({ slug }) => slug);
}
```

Then this effect becomes:

```js
// Get number of most recent templates.
useEffect(() => {
  let mounted = true;

  (async () => {
    const templates = await getRecentTemplateNames(SIXTY_DAYS);
    if (mounted) {
      setNumNewTemplates(templates.length);
    }
  })();

  return () => {
    mounted = false;
  };
}, []);
```

Reply:
yea right now `@web-stories-wp/date` is only used for the tests, so left it as a dev dependency. that being said, if we're fine with moving it to be a dep of the templates package, happy to update to the exact requested api. Can at least mirror the sentiment here either way